### ISSUE TITLE
InconsistentCapitalization warning for names in different scopes

### ISSUE THREAD

**Issue by msridhar** [google/error-prone#1033]
### What version of Error Prone are you using?

Master

### Does this issue reproduce with the latest release?

Yes

### What did you do?

```java
class Foo {
  private String testID;
  static final class Inner {
    Inner(String testId) {}
  }
}
```

### What did you expect to see?

No warning.

### What did you see instead?

```
Foo.java:4: warning: [InconsistentCapitalization] Found the field 'testID' with the same name as the parameter 'testId' but with different capitalization.
    Inner(String testId) {}
                 ^
    (see https://errorprone.info/bugpattern/InconsistentCapitalization.md)
  Did you mean 'Inner(String testID) {}'?
```

According to [the docs](http://errorprone.info/bugpattern/InconsistentCapitalization) the check is supposed to find "two or more variables under the same scope that differ only in capitalization."  The word "under" is a bit ambiguous.  If it means "in the same scope," then the above may be a bug as the names are not in the same scope.  If "under" means something else, might be good to clarify.

**Comment by awturner** [comment]
The second paragraph of the docs clarifies exactly what is checked for:

"This checker will only find parameters that differ in capitalization with fields that can be accessed from the parameter’s scope."

The check is WAI; but perhaps the wording of the documentation could be clarified.

**Comment by msridhar** [comment]
@awturner in this case, I believe the field cannot be accesed from the parameter's scope, as the inner class is static.  But maybe I am misunderstanding.
